src/test/ui/error-codes/E0152.stderr

```diff
@@ -4,7 +4,7 @@ error[E0152]: duplicate lang item found: `arc`.
 LL | struct Foo;
    | ^^^^^^^^^^^
    |
-   = note: first defined in crate `alloc`.
+   = note: first defined in crate `alloc` (which `rustc_std_workspace_core` depends on).
```

Comment:
I'd emit a list instead like so (up to a max number):
```
= note: first defined in crate `alloc`
  note: which `foo` depends upon
  note: which `bar` depends upon
```

Reply:
In order to do that, it'd need an access to the crate dependency graph:

- currently, the emitted error is from a vector of crates in some order, so only works for one crate back;

- for more crates, one will need to know each crate's depth level with respect to `LOCAL_CRATE` -- I couldn't find a query that could return that.

I saw the dependency graph metadata was accessible in https://github.com/rust-lang/rust/blob/master/src/librustc_metadata/locator.rs but I'm not sure how to query that data easily from within `LanguageItemCollector`.

Any suggestions?

Reply:
Not really, let's stick with your approach then =P